contracts/multihop/src/contract.rs

```diff
@@ -0,0 +1,70 @@
+use soroban_sdk::{contract, contractimpl, contractmeta, Address, Env, Vec};
+
+use crate::error::ContractError;
+use crate::storage::{get_liquidity_pool, save_admin, save_liquidity_pool, Pair, Swap};
+
+// Metadata that is added on to the WASM custom section
+contractmeta!(
+    key = "Description",
+    val = "Contract to enable chaining of multiple swap transactions together"
+);
+
+#[contract]
+pub struct Multihop;
+
+pub trait MultihopTrait {
+    fn initialize(
+        env: Env,
+        admin: Address,
+        liquidity_pools: Vec<(Pair, Address)>,
+    ) -> Result<(), ContractError>;
+
+    fn swap(env: Env, operations: Vec<Swap>) -> Result<(), ContractError>;
+}
+
+#[contractimpl]
+impl MultihopTrait for Multihop {
+    fn initialize(
+        env: Env,
+        admin: Address,
+        liquidity_pools: Vec<(Pair, Address)>,
+    ) -> Result<(), ContractError> {
+        save_admin(&env, &admin);
+
+        for lp in liquidity_pools.iter() {
+            let pair = lp.0;
+            let lp_address = lp.1;
+            save_liquidity_pool(&env, pair, lp_address);
+        }
+
+        env.events()
+            .publish(("initialize", "Multihop factory"), admin);
+
+        Ok(())
+    }
+
+    fn swap(env: Env, operations: Vec<Swap>) -> Result<(), ContractError> {
+        for op in operations.iter() {
+            let _lp_address = get_liquidity_pool(
+                &env,
+                Pair {
+                    token_a: op.ask_asset,
+                    token_b: op.offer_asset,
+                },
+            )?;
+
+            // todo swap me using pair::swap?
+            // ``` fn swap(
+            //         env: Env,
+            //         sender: Address,
+            //         sell_a: bool,
+            //         offer_amount: i128,
+            //         belief_price: Option<i64>,
+            //         max_spread_bps: Option<i64>,
+            //     ) -> Result<(), ContractError>;```
+            // if yes, we'll need more arguments to this function
+        }
```

Comment:
All unknowns set as None.
Also this PR is supposed to be just a skeleton, just leave it with `unimplemented`.
But add query for admin.

Reply:
the get_admin_query is in storage.rs, appended _ so the compiler doesn't complain